lib/puppet/provider/package/apt.rb

```diff
@@ -231,4 +237,10 @@ def insync?(is)
     end
     should_range.include?(is_version)
   end
+
+  private
+
+  def should_use_source?
+    @resource[:source]
+  end
```

Comment:
Maybe would be better just to define a source method like:
```ruby
def source
  @source ||= @resource[:source]
end
```

and in `aptitude`
```ruby
def source
  nil
end
```